addons/sale_project_mrp/models/sale_order.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.depends('order_line.product_id', 'order_line.project_id')
+    def _compute_project_ids(self):
+        super()._compute_project_ids()
+        for order in self:
+            projects = order.order_line.mapped('bom_project_ids')
+            order.project_ids |= projects
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    bom_project_ids = fields.Many2many('project.project', help="project\
+                                        generated by services in a kit")
+    is_kit = fields.Boolean('is a kit', compute='_compute_is_kit', help='A\
+                            project/task should be generated if product is\
+                            a kit and said kit contains a service')
+    bom_id = fields.Many2one('mrp.bom', compute='_compute_bom_id', help='Bom related\
```

Comment:
shouldn't this field be better to put it into `sale_mrp`?

Reply:
For the same reason than above, i'd rather have it here so that it's not created until it's necessary